src/plugins/intel_gpu/src/plugin/ops/matmul.cpp

```diff
@@ -166,6 +168,63 @@ static void CreateMatMulOp(Program& p, const std::shared_ptr<ngraph::op::v0::Mat
         // Add actual gemm
         auto alpha = 1.0f;
         auto beta = 0.0f;
+        auto transA = op->get_transpose_a();
+        auto transB = op->get_transpose_b();
+
+        std::array<ngraph::PartialShape, 2> inputShapes{
+            op->get_input_partial_shape(0),
+            op->get_input_partial_shape(1)
+        };
+
+        auto canTransposeInputs = [] (const std::array<ngraph::PartialShape, 2>& shapes, bool transA, bool transB) -> bool {
+            if (!transA && !transB)
+                return false;
+            if (shapes[0].rank().is_dynamic() ||
+                shapes[1].rank().is_dynamic())
+                return false;
```

Comment:
Please add return false when the shapes are dynamic. (not only rank is dynamic) , because we cannot predict what is the actual runtime shape.

Reply:
Self answer ) It is checked by "dim.is_static() in the below condition" so no issue for dyn shape.